src/main/java/seedu/address/model/person/ModuleTakenPredicate.java

```diff
@@ -0,0 +1,40 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import seedu.address.model.tag.NusModule;
+
+/**
+ * Tests if any of a {@code Person}'s {@code Module} matches any of the keywords given.
+ */
+public class ModuleTakenPredicate implements Predicate<Person> {
+    private final List<NusModule> modules;
+
+    public ModuleTakenPredicate(List<NusModule> modules) {
+        this.modules = modules;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        boolean flag = false;
```

Comment:
Should this be named `hasMod` instead?

Reply:
I think this should remain as `test()` in accordance with the overriden method from `Predicate::test()` which both predicates that we have inherit from.